tracing/tracer.go

```diff
@@ -294,6 +305,27 @@ func (t *Tracer) Extract(format interface{}, carrier interface{}) (opentracing.S
 	return nil, opentracing.ErrInvalidCarrier
 }
 
+func (t *Tracer) newID() string {
+	if t.useUUID {
+		id, err := uuid.NewV4()
+		if err != nil {
+			t.logger.Log(
+				context.Background(),
+				fmt.Sprintf("Failed to generate uuid: %v", err),
+			)
+			// On modern linux kernel, after the system entropy pool is initialized
+			// it's guaranteed to be able to get up to 256 bytes in one call without
+			// error [1], so this should never happen.
+			// If this really happens we don't really have a good way to deal with it
+			// anyways.
+			// [1]: https://man7.org/linux/man-pages/man2/getrandom.2.html
+			panic(err)
```

Comment:
I wonder if we should just return empty string in this case? @pacejackson @konradreiche what do you think?

Reply:
It would be interesting to learn about it but this would be outside of the initialization path of the program so I would be definitely for returning an empty string instead of panic.

Reply:
![image](https://user-images.githubusercontent.com/1929108/107691550-6b6e8f00-6c60-11eb-9eb8-18da08694fd4.png)

Agree on it probably being better to return an empty string, I don't think we want initializing spans to potentially cause requests to fail.